Src/Eiffel/eiffel/byte_code/check_b.e

```diff
@@ -99,13 +99,23 @@ feature -- Code generation
 					buf.exdent
 					buf.put_new_line
 					buf.put_character ('}')
-				elseif context.assertion_level.is_check then
-					buf.put_string ("if (~in_assertion) {")
-					buf.indent
-					check_list.generate
-					buf.exdent
-					buf.put_new_line
-					buf.put_character ('}')
+				else
+					check is_final_mode: context.final_mode end
+					if
+						system.keep_assertions and then
+						context.assertion_level.is_check
+					then
+						buf.put_string ("if (~in_assertion) {")
+						buf.indent
+						check_list.generate
+						buf.exdent
+						buf.put_new_line
+						buf.put_character ('}')
+					elseif context.system.exception_stack_managed then
+ 						if attached (create {ASSERTION_BREAKABLE_SLOT_STRATEGY}).breakable_slot_count (check_list, context) as nb then
```

Comment:
if we use a once function, this means it will get harder to get concurrent compilation .... one day.
But ok, I will centralize the call, to make it easier.

Reply:
We are talking about once-per-thread here, so there is no problem with concurrent execution.